src/modules/booking/containers/booking-time-select/booking-time-select.tsx

```diff
@@ -0,0 +1,152 @@
+'use client';
+import { useMemo, type FC } from 'react';
+import clsx from 'clsx';
+import { format } from 'date-fns';
+// utils
+import { trpc } from '@/modules/core/utils/trpc.utils';
+import { generateDates } from '@/modules/core/utils/date.utils';
+import { formatTime } from '@/modules/core/utils/time.utils';
+// components
+import { BookingTimeSelectNavigation } from '../../components/booking-time-select-navigation';
+import { Spinner } from '@/modules/core/components/spinner';
+import { Typography } from '@/modules/core/components/typogrpahy';
+// type
+import type { BookingTimeSelectProps } from '@/modules/booking/containers/booking-time-select/booking-time-select.interface';
+// Swiper components
+import { Swiper, SwiperSlide } from 'swiper/react';
+// Swiper styles
+import 'swiper/scss';
+import 'swiper/scss/navigation';
+// style
+import styles from './booking-time-select.module.scss';
+
+export const BookingTimeSelect: FC<BookingTimeSelectProps> = ({
+  selectedDay,
+  setSelectedDay,
+  selectedTimeRange,
+  setSelectedTimeRange,
+  professionalId,
+}) => {
+  const bookingData = trpc.booking.available.list.useQuery(
+    {
+      date: selectedDay || '',
+      serviceOnProfessionalId: professionalId,
+    },
+    { enabled: !!selectedDay }
+  );
+
+  const handleDayChoose = (date: string) => {
+    setSelectedDay(date);
+
+    setSelectedTimeRange(null);
+  };
+
+  const bookingAvalibleTimes = useMemo(() => {
+    if (bookingData.data) {
+      return bookingData.data;
+    }
+
+    return [];
+  }, [bookingData]);
+
+  return (
+    <div className={styles.root}>
+      <div className={styles.bookingContent}>
+        <Swiper
+          spaceBetween={0}
+          slideNextClass={styles.slideNext}
+          loop={false}
+          slidesPerView={7}
+          className={styles.swiper}
+        >
+          <BookingTimeSelectNavigation selectedDay={selectedDay} />
+
+          {generateDates().map((item, index) => (
```

Comment:
Better to move it to the memo, and then use as const, like `generatedDates.map......`